yarn-project/end-to-end/src/e2e_auth_contract.test.ts

```diff
@@ -0,0 +1,119 @@
+import { AccountWallet, AztecAddress, ContractFunctionInteraction, Fr, PXE } from '@aztec/aztec.js';
+import { AuthContract } from '@aztec/noir-contracts.js';
+
+import { jest } from '@jest/globals';
+
+import { publicDeployAccounts, setup } from './fixtures/utils.js';
+
+describe('e2e_auth_contract', () => {
+  const TIMEOUT = 120_000;
+  jest.setTimeout(TIMEOUT);
+
+  let teardown: () => Promise<void>;
+
+  let admin: AccountWallet;
+  let authorized: AccountWallet;
+  let other: AccountWallet;
+
+  let pxe: PXE;
+
+  let contract: AuthContract;
+
+  const VALUE = 5;
+
+  beforeAll(async () => {
+    ({
+      teardown,
+      wallets: [admin, authorized, other],
+      pxe,
+    } = await setup(3));
+
+    await publicDeployAccounts(admin, [admin.getAddress(), authorized.getAddress(), other.getAddress()]);
+
+    const deployTx = AuthContract.deploy(admin, admin.getAddress()).send({});
+    const receipt = await deployTx.wait();
+    contract = receipt.contract;
+  });
+
+  afterAll(() => teardown());
+
+  async function mineBlock() {
+    await contract.methods.get_authorized().send().wait();
+  }
+
+  async function mineBlocks(amount: number) {
+    for (let i = 0; i < amount; ++i) {
+      await mineBlock();
+    }
+  }
+
+  async function assertLoggedAddress(interaction: ContractFunctionInteraction, address: AztecAddress) {
+    const logs = await pxe.getUnencryptedLogs({ txHash: (await interaction.send().wait()).txHash });
+    expect(AztecAddress.fromBuffer(logs.logs[0].log.data)).toEqual(address);
+  }
+
+  async function assertLoggedNumber(interaction: ContractFunctionInteraction, value: number) {
+    const logs = await pxe.getUnencryptedLogs({ txHash: (await interaction.send().wait()).txHash });
+    expect(Fr.fromBuffer(logs.logs[0].log.data)).toEqual(new Fr(value));
+  }
+
+  it('authorized is unset initially', async () => {
+    await assertLoggedAddress(contract.methods.get_authorized(), AztecAddress.ZERO);
+  });
+
+  it('admin sets authorized', async () => {
+    await contract.withWallet(admin).methods.set_authorized(authorized.getAddress()).send().wait();
+
+    await assertLoggedAddress(contract.methods.get_scheduled_authorized(), authorized.getAddress());
+  });
+
+  it('authorized is not yet set, cannot use permission', async () => {
+    await assertLoggedAddress(contract.methods.get_authorized(), AztecAddress.ZERO);
+
+    await expect(
+      contract.withWallet(authorized).methods.do_private_authorized_thing(VALUE).send().wait(),
+    ).rejects.toThrow('caller is not authorized');
+  });
+
+  it('after a while the scheduled change is effective and can be used with max block restriction', async () => {
+    await mineBlocks(5); // This gets us past the block of change, since the delay is 5 blocks
+
+    const interaction = contract.methods.get_authorized();
+
+    // These assertions are currently failing, need to investigate why.
```

Comment:
Is this fixed?

Reply:
No, trying to figure out what's going on

Reply:
Fixed in https://github.com/AztecProtocol/aztec-packages/pull/5490/commits/f56750791e34c5de131547a28f36a92ffedc9a4e - I was looking at the wrong interaction